### PR TITLE
Bug PM-206 Asked for approval every transaction

### PR DESCRIPTION
**Pull request by mmvsha**
This PR adds a check for the allowance before calling gnosis.js methods. Currently the allowance is set to transaction cost everytime, so we have to approve two transactions: Approve and Buy/Sell. If the allowance is low or the same as transaction cost, it sets it to 2^256 - 1, so we have to approve only one transaction for fairly long time. I can be wrong with allowance term, but this is how I understand it and this is how its done in gnosis.js, except the value it sets allowance to.

**TEST STEPS:**
1. Find a market you never interacted with.
2. Buy shares, you'll be asked to approve an approval transaction.
3. Buy shares multiple times more, you shouldn't be asked to approve an approval transaction

### CODE REVIEW

**Comment by apanizo** [comment]
Can you specify where are the code changes (not the automatic lint fixes) and how can I test this PR (a super short steps description)?

File: src/components/MarketBuySharesForm/index.js [hunk 1]
```diff
@@ -83,20 +89,22 @@ class MarketBuySharesForm extends Component {
...
+      .then(() => {
+        // Fetch new trades
+        this.props.fetchMarketTrades(market)
+        // Fetch new market participant trades
+        this.props.fetchMarketParticipantTrades(market.address, defaultAccount)
+        // Fetch new shares
+        this.props.fetchMarketShares(defaultAccount)
+        return reset()
+      })
+      .catch(e => console.log(e))
```
Comment (andre-meyer) [inline src/components/MarketBuySharesForm/index.js]: `console.log`

File: src/components/MarketBuySharesForm/index.js [hunk 2]
```diff
@@ -126,7 +134,7 @@ class MarketBuySharesForm extends Component {
 
   renderCategorical() {
     const {
-      selectedBuyInvest, selectedOutcome, limitMargin, market, market: { eventDescription }
+      selectedBuyInvest, selectedOutcome, limitMargin, market, market: { eventDescription },
```
Comment (andre-meyer) [inline src/components/MarketBuySharesForm/index.js]: Syntax Error?

File: src/components/MarketMySharesForm/index.js [hunk 3]
```diff
@@ -91,7 +99,11 @@ class MarketMySharesForm extends Component {
     const sellAmount = shareBalanceRounded.sub(selectedSellAmount).lt(MIN_CONSIDER_VALUE)
       ? weiToEth(shareBalance)
       : shareAmount
-    await this.props.sellShares(this.props.market, shareIndex, sellAmount, earnings)
+    try {
+      await this.props.sellShares(this.props.market, shareIndex, sellAmount, earnings)
+    } catch (e) {
+      console.error(e)
```
Comment (andre-meyer) [inline src/components/MarketMySharesForm/index.js]: `console.error`

**Comment by mmvsha** [comment]
@apanizo BuyShares and SellShares methods inside `actions/market.js`

**Comment by apanizo** [comment]
Remember solve the conflicts!

**Pull request merged**